Document TOPLEVEL_LANG (#1924)
cocotb/cocotb: documentation/source/building.rst

    You can usually write out such a file from the simulator's GUI.
    This is currently supported for the Mentor Questa, Mentor ModelSim and Aldec Riviera simulators.

.. make:var:: TOPLEVEL_LANG

    Used to inform the makefile scripts which HDL language the top-level design element is written in.
    Currently it supports the values ``verilog`` for Verilog or SystemVerilog tops, and ``vhdl`` for VHDL tops.
    This is used by simulators that support more than one interface (VPI, VHPI, or FLI) to select the appropriate interface to start cocotb.

    The variable is also made available to cocotb tests conveniently as :data:`cocotb.LANGUAGE`.

>     This can be used to prevent individual tests from running with the selected :make:var:`TOPLEVEL_LANG`.

Reviewer: Perhaps move this sentence to the docs for `cocotb.LANGUAGE`, maybe as:
```
.. info::
    cocotb's own regression tests sometimes run the same tests against both a
    Verilog and VHDL implementation of the same module, where this variable
    can be used to skip tests that only make sense in one of the two languages.
```

Author: I removed the sentence, but I'm hesitant to document the use of `LANGUAGE` because I'm not sure if we should be doing that *at all*. Really tests should run agnostic to the implementation language, that's one of the main points of the GPI. If this is necessary we should be segregating the tests into different suite where in the makefiles we decide whether to skip the whole suite or not.

> 

Reviewer: Should this be right next to `TOP_LEVEL`? This isn't optional, like all the other variables in this section.

Reviewer: Or perhaps vice versa...

Author: Moved to right before `VERILOG_SOURCES`.

    You can usually write out such a file from the simulator's GUI.
    This is currently supported for the Mentor Questa, Mentor ModelSim and Aldec Riviera simulators.

.. make:var:: TOPLEVEL_LANG

>     Used to inform the makefile scripts what HDL language of the toplevel design element is written in.

Reviewer: ```suggestion
    Used to inform the makefile scripts which HDL language the top-level design element is written in.
```

    Currently it supports the values ``verilog`` for Verilog or SystemVerilog tops, and ``vhdl`` for VHDL tops.
    This is used by simulators that support more than one interface (VPI, VHPI, or FLI) to select the appropriate interface to start cocotb.

>     It is also commonly used to deter execution of a cocotb test suite if the tests or DUT do not support being run with the selected :make:var:`TOPLEVEL_LANG`.

Reviewer: This doesn't seem like user information - is anything other than cocotb itself going to have two copies of their code, one vhdl and one verilog?

Author: Potentially? Really the first few sentences are the pertinent information and this just follows. I'm not sure where else to document how to run the regression with different langs. 

Speaking of which, I just realized we don't have TOPLEVEL_LANG exempted in tox.

Reviewer: That probably should go in the contribution guide, for how to run cocotb's own unit tests. I think only the first three sentences are useful here.

I suppose you could put the remaining sentences in an `.. info` box, with a comment that this is something that cocotb does in its own test-suite, not something that users actually ought to do.

Author: I'll add a section to the CONTRIBUTING.md about it as well as `SIM` being supported by `tox` once the `TOPLEVEL_LANG` exemption is added to tox.ini.

>     The variable is also made available to cocotb tests conveniently at :data:`cocotb.LANGUAGE`.

Reviewer 2: s/at/as/?
